File: src/gestorAplicacion/artefactos/Objeto.java
/*
Se implementa para la creación de los objetos que van a estar presentes en el juego


Autores: Camilo, Yeferson 
*/

package artefactos;
import java.io.Serializable;
import java.util.ArrayList;

import complementario.Habilidad;
import complementario.TipoDeObjeto;

public class Objeto implements Serializable{
	
	private static final long serialVersionUID = 1L;
	
	private final TipoDeObjeto tipo;  //Clasificación del objeto segun su tipo
	private final String nombre;      //Nombre del objeto
	private final int precio;	      //Constante con el valor del objeto
	private int ataqueFisico;         //Contribución al ataque físico de quien lo posee
	private int ataqueMagico;         //Contribución al ataque mágico de quien lo posee
	private int defensaFisica;        //Contribución al defensa física de quien lo posee
	private int defensaMagica;        //Contribución al defensa mágica de quien lo posee
	private Habilidad habilidad;      //Habilidad del objeto
	private int durabilidad;          //Numero de turnos que dura el objeto sin deteriorarse
	private boolean evoluciona;       //Determina si el objeto puede o no evolucionar
	private Objeto evolucion;         //Objeto al que evoluciona
	private ArrayList<Objeto> materialesEvolucion = new ArrayList<Objeto>();  //materiales necesarios para evolucionar
	
	
	// Constructor armas,armaduras y anillos
	public Objeto(String nombre,int precio,TipoDeObjeto tipo,int ataqueFisico,int ataqueMagico,int defensaFisica,int defensaMagica,int durabilidad,boolean evoluciona) {
		this.nombre = nombre;
		this.precio = precio;
		this.tipo = tipo;
		this.ataqueFisico = ataqueFisico;
		this.ataqueMagico = ataqueMagico;
		this.defensaFisica = defensaFisica;
		this.defensaMagica = defensaMagica;
		this.durabilidad = durabilidad;
		this.evoluciona = evoluciona;
	}
	
	// Constructor pociones
	public Objeto(String nombre,int precio,Habilidad habilidad) {
		this(nombre,precio,TipoDeObjeto.POCION,0,0,0,0,1,false);
		this.habilidad = habilidad;
	
	}
	
	// Constructor de materiales
	public Objeto(String nombre) {
		this(nombre,0,TipoDeObjeto.MATERIAL,0,0,0,0,1,false);
		
	}
	

	public TipoDeObjeto getTipo() {
		return tipo;
	}

	public int getPrecio() {
		return precio;
	}


	public int getAtaqueFisico() {
		return ataqueFisico;
	}

	public void setAtaqueFisico(int ataqueFisico) {
		this.ataqueFisico = ataqueFisico;
	}

	public int getAtaqueMagico() {
		return ataqueMagico;
	}

	public void setAtaqueMagico(int ataqueMagico) {
		this.ataqueMagico = ataqueMagico;
	}

	public Habilidad getHabilidad() {
		return habilidad;
	}

	public void setHabilidad(Habilidad habilidad) {
		this.habilidad = habilidad;
	}

	public int getDurabilidad() {
		return durabilidad;
	}

	public void setDurabilidad(int durabilidad) {
		this.durabilidad = durabilidad;
	}

	public String getNombre() {
		return nombre;
	}
	
	public boolean isEvoluciona() {
		return evoluciona;
	}
	
	public Objeto getEvolucion() {
		return evolucion;
	}
	

	public int getDefensaFisica() {
		return defensaFisica;
	}

	public void setDefensaFisica(int defensaFisica) {
		this.defensaFisica = defensaFisica;
	}

	public int getDefensaMagica() {
		return defensaMagica;
	}

	public void setDefensaMagica(int defensaMagica) {
		this.defensaMagica = defensaMagica;
	}

	public void setEvoluciona(boolean evoluciona) {
		this.evoluciona = evoluciona;
	}

	public void setEvolucion(Objeto evolucion) {
		this.evolucion = evolucion;
	}

	public String toString() {
		return "- " + getNombre() + " : " + getAtaqueFisico() + ", " + getAtaqueMagico()
				+ ", " + getDefensaFisica() + ", " + getDefensaMagica() + ", " + getDurabilidad() ;
				
	}

	public ArrayList<Objeto> getMaterialesEvolucion() {
		return materialesEvolucion;
	}

	public void setMaterialesEvolucion(ArrayList<Objeto> materialesEvolucion) {
		this.materialesEvolucion = materialesEvolucion;
	}
	
	
	
	
	
	
	
	
}

File: Python/src/gestorAplicacion/artefactos/objeto.py
'''
Se implementa para la creación de los objetos que van a estar presentes en el juego


Autores: Camilo, Yeferson, Edwin
'''



from tkinter import PhotoImage
from turtle import width


class Objeto:

    def __init__(self, nombre, habilidad = None, precio = 0, tipo = 0, ataqueFisico = 0, ataqueMagico = 0, defensaFisica = 0, defensaMagica = 0, durabilidad = 0, evoluciona = False,icono = None):
        self._nombre = nombre
        self._precio = precio
        self._tipo = tipo
        self._ataqueFisico = ataqueFisico
        self._ataqueMagico = ataqueMagico
        self._defensaFisica = defensaFisica
        self._defensaMagica = defensaMagica
        self._durabilidad = durabilidad
        self._evoluciona = evoluciona
        self._hechizo = habilidad
        self._evolucion = None
        self._materialesEvolucion = None
        self._icono = icono
            


    def getTipo(self):
        return self._tipo

    def getPrecio(self):
        return self._precio

    def getIcono(self):
        return self._icono

    def setIcono(self,icono):
        self._icono = icono

    def getAtaqueFisico(self):
        return self._ataqueFisico

    def setAtaqueFisico(self, ataqueFisico):
        self._ataqueFisico = ataqueFisico

    def getAtaqueMagico(self):
        return self._ataqueMagico   

    def setAtaqueMagico(self, ataqueMagico):
        self._ataqueMagico = ataqueMagico

    def setHechizo(self,hechizo):
        self._hechizo = hechizo

    def getHechizo(self):
        return self._hechizo       

    def getDurabilidad(self):
        return self._durabilidad

    def setDurabilidad(self, durabilidad):
        self._durabilidad = durabilidad

    def getNombre(self):
        return self._nombre

    def isEvoluciona(self):
        return self._evoluciona

    def getEvolucion(self):
        return self._evolucion


    def getDefensaFisica(self):
        return self._defensaFisica

    def setDefensaFisica(self, defensaFisica):
        self._defensaFisica = defensaFisica

    def getDefensaMagica(self):
        return self._defensaMagica

    def setDefensaMagica(self, defensaMagica):
        self._defensaMagica = defensaMagica


    def setEvolucion(self, evolucion):
        self._evolucion = evolucion

    def __str__(self):
        texto = "- " + self.getNombre() + " : " + str(self.getAtaqueFisico()) + ", " + str(self.getAtaqueMagico()) + ", " + str(self.getDefensaFisica()) + ", " + str(self.getDefensaMagica()) + ", " + str(self.getDurabilidad())
        return texto

    def getMaterialesEvolucion(self):
        return self._materialesEvolucion

    def setMaterialesEvolucion(self, materialesEvolucion):
        self._materialesEvolucion = materialesEvolucion









